web/src/app/chat/components/WelcomeMessage.tsx

```diff
@@ -16,10 +19,15 @@ export default function WelcomeMessage({
   agent,
   isDefaultAgent,
 }: WelcomeMessageProps) {
+  const [greeting, setGreeting] = useState(() => GREETING_MESSAGES[0] ?? "");
```

Comment:
Nice! You're extracting the hook out into an unconditional pathway.

I'm curious as to why this wasn't done before. If the `git blame` points to me, it's probably because I move the file and formatted it a bit.